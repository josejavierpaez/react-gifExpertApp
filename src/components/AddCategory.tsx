import React, { useState } from 'react'

interface Props {
  setCategories: Function;
}

export const AddCategory = ({ setCategories, }: Props) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e: React.ChangeEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats: string[]) => [inputValue, ...cats]);
      setInputValue('');
    }
  }
  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={handleInputChange}
          required={true}
        />
      </form>
    </>
  )
}
