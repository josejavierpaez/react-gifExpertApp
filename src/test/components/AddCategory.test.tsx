import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory';

describe('test to AddCategory component', () => {

  const setCategories = jest.fn();
  let wraper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wraper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should be render AddCategory component well', () => {

    expect(wraper).toMatchSnapshot();
  })

  test('should be change textbox', () => {
    const input = wraper.find('input');
    const value = 'hellow world';

    input.simulate('change', {
      target: { value }
    });
  })
  test('should not be post information when submit the form', () => {

    wraper.find('form').simulate('submit', {
      preventDefault() { }
    });

    expect(setCategories).not.toHaveBeenCalled();
  })

  // test('should be call setCategories and clear the textbox', () => {

  //   const input = wraper.find('input');
  //   const value = 'hello world';
  //   input.simulate('change', {
  //     target: { value }
  //   });

  //   wraper.find('form').simulate('submit', {
  //     preventDefault() { }

  //   })

  //   expect(setCategories).toHaveBeenCalled();
  //   expect(setCategories).toHaveBeenCalledTimes(1);
  //   expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
  //   expect(wraper.find('input').prop('value')).toBe('');
  // })



});