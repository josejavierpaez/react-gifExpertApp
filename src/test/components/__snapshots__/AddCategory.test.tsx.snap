// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to AddCategory component should be render AddCategory component well 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
